lib/util/viewer.js

```diff
@@ -185,3 +157,35 @@ export function getModeFromRoute (route) {
   }
   return route.mode || modeLookup[route.type]
 }
+
+/**
+ * Enum to represent transit realtime status for trips/stop times.
+ */
+export const REALTIME_STATUS = {
+  EARLY: 'EARLY',
+  LATE: 'LATE',
+  ON_TIME: 'ON_TIME',
+  SCHEDULED: 'SCHEDULED'
+}
+
+/**
+ * Get one of the realtime states (on-time, late...) if a leg/stoptime is
+ * registering realtime info and given a delay value in seconds.
+ */
+export function getTripStatus (isRealtime, delaySeconds) {
+  if (isRealtime) {
+    if (delaySeconds > 60) {
+      // late departure
+      return REALTIME_STATUS.LATE
+    } else if (delaySeconds < -60) {
+      // early departure
+      return REALTIME_STATUS.EARLY
+    } else {
+      // on-time departure
+      return REALTIME_STATUS.ON_TIME
+    }
```

Comment:
Consider basing this PR off of #305 and then refactor this function so that it uses the configurable on-time threshold.

Reply:
Actually, since #305 was introduced after the latest commit here, I think it might be unfair to request this. Therefore, I'm going to merge this and make #305 dependent on this.